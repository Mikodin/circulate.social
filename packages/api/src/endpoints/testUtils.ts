import { APIGatewayProxyEvent, Context } from 'aws-lambda';

export function CreateMockEvent(event): APIGatewayProxyEvent {
  const mockEvent = {
    requestContext: {
      authorizer: {
        clams: {
          'cognito:username': 'Dev User',
          // eslint-disable-next-line @typescript-eslint/camelcase
          email_verified: true,
        },
      },
    },
    ...event,
    body: JSON.stringify(event.body || {}),
  };

  return mockEvent;
}

export function CreateMockContext(context): Context {
  return {
    ...context,
  };
}
