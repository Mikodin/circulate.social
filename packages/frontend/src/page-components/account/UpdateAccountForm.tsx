import { useState } from 'react';
import { Form, Input, Select, Button } from 'antd';
import { ZoneId } from '@js-joda/core';
import Axios from 'axios';
import '@js-joda/timezone';

import { UserContextType } from '../../state-management/UserContext';
import { API_ENDPOINT } from '../../util/constants';

export const AVAILABLE_TIMEZONES = ZoneId.getAvailableZoneIds().sort();
const TimezoneSelects = AVAILABLE_TIMEZONES.map((timeZone) => (
  <Select.Option value={timeZone} key={timeZone}>
    {timeZone}
  </Select.Option>
));

const UPDATE_ACCOUNT_ENDPOINT = `${API_ENDPOINT}/user/edit`;

type Props = {
  user: UserContextType['user'];
  jwtToken: string;
  refreshUser: (bypassCache: boolean) => Promise<UserContextType['user']>;
};

interface FormValues {
  firstName: string;
  lastName: string;
  timezone: string;
}

const UpdateAccountForm = ({
  user,
  jwtToken,
  refreshUser,
}: Props): JSX.Element => {
  const { firstName, lastName, timezone } = user;
  const [form] = Form.useForm();
  const [isUpdateAccountInFlight, setIsUpdateAccountInFlight] = useState(false);

  const onFormFinish = async (values: FormValues) => {
    const userValues = {
      firstName: values.firstName,
      lastName: values.lastName,
      timezone: values.timezone,
    };
    setIsUpdateAccountInFlight(true);
    await Axios.patch(
      UPDATE_ACCOUNT_ENDPOINT,
      { ...userValues },
      {
        headers: { Authorization: jwtToken },
      }
    );

    await refreshUser(true);
    setIsUpdateAccountInFlight(false);

    return values;
  };

  return (
    <div>
      <Form
        className={''}
        form={form}
        name="horizontal_login"
        size="large"
        layout="vertical"
        onFinish={onFormFinish}
        initialValues={
          {
            firstName,
            lastName,
            timezone,
          } as FormValues
        }
      >
        <Form.Item
          name="firstName"
          rules={[
            { required: true, message: 'Sorry, you must have your first name' },
          ]}
          label="First Name"
        >
          <Input placeholder="First name" type="text" />
        </Form.Item>
        <Form.Item
          name="lastName"
          rules={[
            { required: true, message: 'Sorry, you must have your last name' },
          ]}
          label="Last Name"
        >
          <Input placeholder="Last name" type="text" />
        </Form.Item>
        <Form.Item name="timezone" label="Timezone">
          <Select showSearch style={{ width: 200 }} placeholder="Timezone">
            {TimezoneSelects}
          </Select>
        </Form.Item>
        <Form.Item shouldUpdate={true}>
          {(): JSX.Element => (
            <Button
              data-testid="submitButton"
              loading={isUpdateAccountInFlight}
              type="primary"
              htmlType="submit"
              size="large"
              block
              disabled={Boolean(
                form.getFieldsError().filter(({ errors }) => errors.length)
                  .length
              )}
            >
              Update
            </Button>
          )}
        </Form.Item>
      </Form>
    </div>
  );
};

export default UpdateAccountForm;
