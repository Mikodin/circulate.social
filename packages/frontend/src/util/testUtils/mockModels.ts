import { Circle, User } from '@circulate/types';

export const mockCircle = {
  id: 'asdf-someCircle-id',
  createdAt: '123',
  updatedAt: '123',
  members: ['dev-id'],
  createdBy: 'dev-id',
  name: 'The greatest circle',
  description: 'Great things happen here',
  frequency: 'daily',
  privacy: 'private',
} as Circle;

export const mockUser = {
  id: 'some_user',
  email: 'billNye@circulate.social',
  firstName: 'Bill',
  lastName: 'Nye',
  timezone: 'UTC',
  createdAt: '123',
  updatedAt: '13',
} as User;
