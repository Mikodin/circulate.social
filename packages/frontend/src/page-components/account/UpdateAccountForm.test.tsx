import { render } from '@testing-library/react';
// import axios from 'axios';

import UpdateAccountForm, { Props } from './UpdateAccountForm';
import { mockUser } from '../../util/testUtils/mockModels';

const DEFAULT_PROPS: Props = {
  user: mockUser,
  jwtToken: 'some_token',
  refreshUser: () => Promise.resolve(mockUser),
};
function renderUpdateAccountForm(overrideProps?: Partial<Props>) {
  return render(<UpdateAccountForm {...DEFAULT_PROPS} {...overrideProps} />);
}

describe('UpdateAccountForm', () => {
  it('Should exist', () => {
    expect(renderUpdateAccountForm()).toMatchSnapshot();
  });
});
