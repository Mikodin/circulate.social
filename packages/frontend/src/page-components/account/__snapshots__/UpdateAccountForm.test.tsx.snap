// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateAccountForm Should exist 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form
          class="ant-form ant-form-vertical ant-form-large"
          id="horizontal_login"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label"
            >
              <label
                class="ant-form-item-required"
                for="horizontal_login_firstName"
                title="First Name"
              >
                First Name
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input ant-input-lg"
                    id="horizontal_login_firstName"
                    placeholder="First name"
                    type="text"
                    value="Bill"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label"
            >
              <label
                class="ant-form-item-required"
                for="horizontal_login_lastName"
                title="Last Name"
              >
                Last Name
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input ant-input-lg"
                    id="horizontal_login_lastName"
                    placeholder="Last name"
                    type="text"
                    value="Nye"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label"
            >
              <label
                class=""
                for="horizontal_login_timezone"
                title="Timezone"
              >
                Timezone
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-select ant-select-lg ant-select-single ant-select-show-arrow ant-select-show-search"
                    style="width: 200px;"
                  >
                    <div
                      class="ant-select-selector"
                    >
                      <span
                        class="ant-select-selection-search"
                      >
                        <input
                          aria-activedescendant="horizontal_login_timezone_list_0"
                          aria-autocomplete="list"
                          aria-controls="horizontal_login_timezone_list"
                          aria-haspopup="listbox"
                          aria-owns="horizontal_login_timezone_list"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="horizontal_login_timezone"
                          role="combobox"
                          type="search"
                          value=""
                        />
                      </span>
                      <span
                        class="ant-select-selection-item"
                        title="UTC"
                      >
                        UTC
                      </span>
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-arrow"
                      style="user-select: none;"
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        class="anticon anticon-down ant-select-suffix"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-lg ant-btn-block"
                    data-testid="submitButton"
                    type="submit"
                  >
                    <span>
                      Update
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form
        class="ant-form ant-form-vertical ant-form-large"
        id="horizontal_login"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="horizontal_login_firstName"
              title="First Name"
            >
              First Name
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input ant-input-lg"
                  id="horizontal_login_firstName"
                  placeholder="First name"
                  type="text"
                  value="Bill"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="horizontal_login_lastName"
              title="Last Name"
            >
              Last Name
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input ant-input-lg"
                  id="horizontal_login_lastName"
                  placeholder="Last name"
                  type="text"
                  value="Nye"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class=""
              for="horizontal_login_timezone"
              title="Timezone"
            >
              Timezone
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-select ant-select-lg ant-select-single ant-select-show-arrow ant-select-show-search"
                  style="width: 200px;"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="horizontal_login_timezone_list_0"
                        aria-autocomplete="list"
                        aria-controls="horizontal_login_timezone_list"
                        aria-haspopup="listbox"
                        aria-owns="horizontal_login_timezone_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="horizontal_login_timezone"
                        role="combobox"
                        type="search"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="UTC"
                    >
                      UTC
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  class="ant-btn ant-btn-primary ant-btn-lg ant-btn-block"
                  data-testid="submitButton"
                  type="submit"
                >
                  <span>
                    Update
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
